server/conn.go

```diff
@@ -1316,6 +1316,18 @@ func (cc *clientConn) handleIndexAdvise(ctx context.Context, indexAdviseInfo *ex
 func (cc *clientConn) handleQuery(ctx context.Context, sql string) (err error) {
 	defer trace.StartRegion(ctx, "handleQuery").End()
 	sc := cc.ctx.GetSessionVars().StmtCtx
+	detachSessionTracker := func() {
```

Comment:
I do not think it's a good idea to handle this only on `handleQuery`, the other protocol like `handleStmtExecute` may be missed.

Reply:
good catch